### ISSUE TITLE
https-proxy-agent vulnerability npm audit

### ISSUE THREAD

**Issue by wilau2** [microsoft/react-native-code-push#1703]
### Steps to Reproduce
1. npm audit

https://www.npmjs.com/advisories/1184

### Environment

* react-native-code-push version: 5.7.0

**Comment by luisnaranjo733** [comment]
Any updates on this? According to the comments on the [hacker one report](https://hackerone.com/reports/541502) that kicked off these flags, I believe that fixing this will not be a breaking change for react-native-code-push

react-native-code-push needs to upgrade to a version of superagent-proxy that depends on the patched up https-proxy-agent@3 (not https-proxy-agent@2).

For reference, these are the two problematic branches in the dependency graph introduced by react-native-code-push@latest

# Path 1
react-native-code-push -> code-push -> superagent-proxy -> proxy-agent ->pac-proxy-agent -> https-proxy-agent@2

# Path 2
react-native-code-push -> code-push -> superagent-proxy -> proxy-agent -> https-proxy-agent@2

**Comment by luisnaranjo733** [comment]
In order to fix this,  react-native-code-push needs code-push to publish a fix, which needs on superagent-proxy to publish a fix.

This is the order in which things need to happen
* [superagent-proxy](https://github.com/TooTallNate/superagent-proxy) needs to land [this PR](https://github.com/TooTallNate/superagent-proxy/pull/35) to close this issue and then publish a new version
* [code-push](https://github.com/microsoft/code-push/) needs to open a new PR that updates to the new version of superagent-proxy so that they can close [this issue](https://github.com/microsoft/code-push/issues/671).
* react-native-code-push needs to open a new PR that updates to the new version of code-push so that they can close (this) issue.

The main blocker right now is [this](https://github.com/TooTallNate/superagent-proxy/issues/34) issue in superagent-proxy.

**Comment by luisnaranjo733** [comment]
On closer inspection, I found that this vulnerability has already been fixed downstream. Coincidentally, the downstream dependency graph has loose enough version targeting that "code-push" and "react-native-code-push" both benefit from the fix for free, without needing a code change.

That being said, consumers of react-native-code-push and code-push that installed those packages before the fix and use lock files are still impacted until they refresh their dependency graph.

In order to upgrade from the vulnerable https-proxy-agent@2 to the patched https-proxy-agent@3, all I needed to do was the following

```
npm uninstall react-native-code-push
npm install react-native-code-push
```

This refreshed my package-lock to a healthy state.
Yarn users can also reinstall to fix their yarn.lock file.

FYI @wilau2
